src/main/java/net/mcreator/ui/modgui/BlockGUI.java

```diff
@@ -552,8 +552,7 @@ public BlockGUI(MCreator mcreator, ModElement modElement, boolean editingMode) {
 
 		blockBase.addActionListener(e -> {
 			if (blockBase.getSelectedItem() != null) {
-				texturesPanelLayout.show(texturesPanel,
-						blockBase.getSelectedItem().equals("Sign") ? "sign" : "normal");
+				texturesPanelLayout.show(texturesPanel, blockBase.getSelectedItem().equals("Sign") ? "sign" : "normal");

```

Comment:
You can just execute the following in this listener (instead of all it has currently):
```java
			if (blockBase.getSelectedItem() != null && blockBase.getSelectedItem().equals("Sign")) {
				signTexture.setValidator(new TileHolderValidator(signTexture));
				texture.setValidator(null);
				texturesPanelLayout.show(texturesPanel, "sign");
			} else {
				signTexture.setValidator(null);
				texturesPanelLayout.show(texturesPanel, "normal");
				texture.setValidator(new TileHolderValidator(texture));
			}
```
This would remove the need for the conditional validators you use later.